File: src/main/java/factorial/Factorial.java
package factorial;

public class Factorial {
    public static void main(String[] args) {
        System.out.println(factorialRecursive(5));
        System.out.println(factorial(5));
    }

    public static int factorialRecursive(int number) {
        return (number == 1) ? 1 : number * factorialRecursive(number - 1);
    }

    public static int factorial(int number) {
        int result = number;
        int currentElement = number - 1;
        while (currentElement != 1) {
            result *= currentElement;
            currentElement--;
        }
        return result;
    }
}

File: src/main/java/factorial/factorial.py
def factorial_recursive(number):
    if number == 1:
        return 1
    else:
        return number * factorial_recursive(number - 1)


def factorial(number):
    result = number
    current_element = number - 1
    while current_element != 1:
        result *= current_element
        current_element -= 1
    return result


print(factorial_recursive(5))
print(factorial(5))
